AirportTicketBookingSystem/AirportTicketBookingSystem/Repositories/FlightRepository.cs

```diff
@@ -0,0 +1,175 @@
+
+            return flights;
+        }
+
+        public void AddFlight(Flight flight)
+        {
+            if (flight == null)
+                throw new ArgumentNullException(nameof(flight));
+
+            var records = GetAllFlights();
+            flight.Id = IdGenerator.GenerateNewId(records);
+
+            records.Add(flight);
+            Save(records);
+        }
+
+        public void UpdateFlight(int flightId, Flight newFlight)
+        {
+            if (newFlight is null) throw new ArgumentNullException(nameof(newFlight));
+
+            var records = GetAllFlights();
+            var index = records.FindIndex(f => f.Id == flightId);
+
+            if (index < 0)
+                throw new KeyNotFoundException("Flight not found to update");
+
+            records[index] = newFlight;
+            Save(records);
+        }
+
+        private void Save(List<Flight> flights)
+        {
+            using var writer = new StreamWriter(_filePath);
+            using var csv = new CsvWriter(writer, CultureInfo.InvariantCulture);
+            csv.WriteRecords(flights);
+        }
+
+        public List<Flight> SearchAvailableFlights(string departureCountry = null, string destinationCountry = null,
+            string departureAirport = null, string arrivalAirport = null, DateTime? departureDateFrom = null,
+            DateTime? departureDateTo = null, TravelClass? seatClass = null, decimal? minPrice = null, decimal? maxPrice = null)
+        {
+            var allFlights = GetAllFlights();
+            var allBookings = _bookingRepository.GetAllBookings();
+
+            FlightBookingUtilities.AttachBookings(allFlights, allBookings);
+
+            var query =
+            from f in allFlights
+            where (string.IsNullOrEmpty(departureCountry) || f.DepartureCountry.Equals(departureCountry, StringComparison.OrdinalIgnoreCase))
+            where (string.IsNullOrEmpty(destinationCountry) || f.DestinationCountry.Equals(destinationCountry, StringComparison.OrdinalIgnoreCase))
+            where (string.IsNullOrEmpty(departureAirport) || f.DepartureAirport.Equals(departureAirport, StringComparison.OrdinalIgnoreCase))
+            where (string.IsNullOrEmpty(arrivalAirport) || f.ArrivalAirport.Equals(arrivalAirport, StringComparison.OrdinalIgnoreCase))
+            where (!departureDateFrom.HasValue || f.DepartureDate >= departureDateFrom.Value)
+            where (!departureDateTo.HasValue || f.DepartureDate <= departureDateTo.Value)
+            where (!seatClass.HasValue || f.Bookings.Any(b => b.SeatClass.Name == seatClass.Value))
+            where (!minPrice.HasValue || f.Bookings.Any(b => b.SeatClass.CalculatePrice() >= minPrice.Value))
+            where (!maxPrice.HasValue || f.Bookings.Any(b => b.SeatClass.CalculatePrice() <= maxPrice.Value))
+            select f;
+
+            return query.ToList();
+        }
+
+        public List<string> ImportFlightsFromCsv(string csvFilePath)
```

Comment:
i would move this to the service layer, and from there i call the db to save the records. Repos really just for doing straightforward db operations, any business logic should live in the service/domain layer

Reply:
treat this as an external importer from csv files, i know we have db saved in csv files, but later it will be an actual db and you will need to import things to it from different resources. You can abstract csv main operations into a helper that can be re-used from repos and services.